ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake

```diff
@@ -27,18 +32,35 @@
 # @public
 #
 macro(ament_auto_find_build_dependencies)
-  if(ARGN)
+  cmake_parse_arguments(ARG "" "" "REQUIRED_PACKAGES" ${ARGN})
```

Comment:
Not considering functions (thanks for pointing that out) it might just be https://github.com/ament/ament_cmake/blob/7b31d859f3d42b9eb6c84db807d95a5604a6881b/ament_cmake_auto/cmake/ament_auto_add_library.cmake#L36 

https://github.com/ament/ament_cmake/blob/36d5c721f5d9b6242126175e9002990017f849b0/ament_cmake_export_libraries/cmake/ament_export_library_names.cmake#L33

and https://github.com/ament/ament_cmake/blob/7b31d859f3d42b9eb6c84db807d95a5604a6881b/ament_cmake_auto/cmake/ament_auto_add_executable.cmake#L36